src/runtime/program/compute_budget/lib.zig

```diff
@@ -109,8 +111,169 @@ const ComputeBudgetInstruction = union(enum(u32)) {
 
     /// Set a specific transaction-wide account data size limit, in bytes, is allowed to load.
     set_loaded_accounts_data_size_limit: u32,
+
+    pub const Tag = @typeInfo(ComputeBudgetInstruction).@"union".tag_type.?;
+
+    const borsh = BorshStaticTaggedUnionHelper(ComputeBudgetInstruction);
+
+    pub const BorshDecodeError = error{
+        /// The raw data did not match the expected representation.
+        DecodeFailed,
+        /// `src` was too short.
+        TruncatedSrc,
+    };
+
+    /// Decodes the encoded ComputeBudgetInstruction, and the number of bytes it occupied in `src`.
+    pub fn borshDecodeSlice(
+        src: []const u8,
+    ) BorshDecodeError!struct { ComputeBudgetInstruction, borsh.EncodedSizeInt } {
+        if (src.len == 0) return error.TruncatedSrc;
+        const tag = borsh.tagDecode(src[0]) orelse return error.DecodeFailed;
+        const pl_size: borsh.EncodedSizeInt = borsh.payloadEncodedSize(tag);
+        const pl_bytes = src[1..];
+        if (pl_bytes.len < pl_size) return error.TruncatedSrc;
```

Comment:
Couldn't you check this first and skip the 0 check?

Reply:
I'm confused what you mean, need to check `src.len == 0` first in order to access `src[0]`.